core/cas-server-core-monitor/src/main/java/org/apereo/cas/monitor/config/CasCoreMonitorConfiguration.java

```diff
@@ -60,4 +66,12 @@ public HealthIndicator sessionHealthIndicator() {
         }
         return () -> Health.up().build();
     }
+
+    @ConditionalOnMissingBean(name = "systemHealthIndicator")
+    @Bean
+    @ConditionalOnEnabledHealthIndicator("systemHealthIndicator")
+    public HealthIndicator systemHealthIndicator() {
+        val warnLoad = casProperties.getMonitor().getLoad().getWarn();
+        return new SystemMonitorHealthIndicator(metricsEndpoint.getIfAvailable(), warnLoad.getThreshold());
```

Comment:
This will cause an NPE if the metrics endpoint is turned off. You should conditionally create this, based on whether metrics is also enabled as an endpoint.